### PR TITLE
Feature/plotting

### PR DESCRIPTION
**Pull request by marksantcroos**
Despite the name this turned out to be more of an IPDPS branch.
So agent profiling changes and general agent improvements towards support for Cray.

### CODE REVIEW

File: src/radical/pilot/agent/radical-pilot-agent-multicore.py [hunk 1]
```diff
@@ -2038,7 +2045,7 @@ def construct_command(self, task_exec, task_args, task_numcores,
         hosts_string = ",".join([slot.split(':')[0] for slot in task_slots])
         export_vars  = ' '.join(['-x ' + var for var in self.EXPORT_ENV_VARIABLES if var in os.environ])
 
-        orte_command = '%s --hnp "%s" %s -np %s -host %s %s' % (
+        orte_command = '%s --debug-devel --hnp "%s" %s -np %s -host %s %s' % (
             self.launch_command, dvm_uri, export_vars, task_numcores, hosts_string, task_command)
```
Comment (andre-merzky) [inline src/radical/pilot/agent/radical-pilot-agent-multicore.py]: should we make the debug dependent on the pilot debug level (which is in `self._cfg['debug']`)?

Comment (marksantcroos) [inline src/radical/pilot/agent/radical-pilot-agent-multicore.py]: We could, although the debug level itself is no longer an option.

Comment (andre-merzky) [inline src/radical/pilot/agent/radical-pilot-agent-multicore.py]: debug is  synced with the debug level on module side:
https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/controller/pilot_launcher_worker.py#L616

Comment (marksantcroos) [inline src/radical/pilot/agent/radical-pilot-agent-multicore.py]: Pfff https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/bootstrapper/default_bootstrapper.sh#L1324

Comment (andre-merzky) [inline src/radical/pilot/agent/radical-pilot-agent-multicore.py]: Yeah, those should go into the SAGA job environment I guess...

**Pull request merged**